### PR TITLE
Styler updates + Layman wfs fix

### PR DESCRIPTION
**Pull request by FilipLeitner**
## Description
[Specific UI for uppermost filter addition](https://github.com/hslayers/hslayers-ng/pull/4330/commits/2bc6af34fdda4ff1555d2d0d15fe9f8d2c17766d)
image

## Related issues or pull requests

#4333 

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [x] Feature
- [ ] Dependency updates
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [ ] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [ ] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

**Comment by FilipLeitner** [comment]
color-map-picker


In IDW widget its  was bit awkward so I just limited the height .
color-map-picker-idw
image

File: projects/hslayers/src/components/styles/styler.service.ts [hunk 1]
```diff
@@ -751,4 +783,111 @@ export class HsStylerService {
...
+    const rect = document.createElementNS('http://www.w3.org/2000/svg', 'rect');
+    rect.setAttribute('width', forLayer ? '50%' : '100%');
+    rect.setAttribute('height', '100%');
+    rect.setAttribute('fill', `url('#${id}')`);
+
+    //Gradient def generation
+    if (input instanceof Layer) {
+      const source = input.getSource();
+      const arr = Array.from(Array(100).keys());
+      for (const i of arr.filter((e, i) => i % 5 === 5 - 1).reverse()) {
```
Comment (jmacura) [inline projects/hslayers/src/components/styles/styler.service.ts:849]:
```suggestion
      for (const i of arr.filter((e, idx) => idx % 5 === 4).reverse()) {
```

`5 - 1` may have sense if you intend to make a variable from 5.
I propose to rename the internal `i` as it is used twice in the same line but in different context.

File: projects/hslayers/src/components/styles/styler.service.ts [hunk 2]
```diff
@@ -751,4 +783,111 @@ export class HsStylerService {
...
+    svg.setAttribute('width', forLayer ? '75%' : '100%');
+    svg.setAttribute('height', '100%');
+
+    const defs = document.createElementNS('http://www.w3.org/2000/svg', 'defs');
+    const linearGradient = document.createElementNS(
+      'http://www.w3.org/2000/svg',
+      'linearGradient',
+    );
+
+    const id = `${+new Date()}-idwGradient-${
```
Comment (jmacura) [inline projects/hslayers/src/components/styles/styler.service.ts:828]: interesting approach as Math.random() is already based on a pseudo-random generation based on a current date 🙂

**Comment by jmacura** [comment]
Ah, sorry, I didn't realize that this PR was meant to be merged into the hot-fix-filip branch. So I will merge into develop, but there are conflict now... 😢

File: projects/hslayers/src/components/styles/styler.service.ts (hunk 1 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/styles/styler.service.ts:849]: This ones interesting :D I cannot quite figure out how come I ended up with 5-1 :D . Probably was just thinking about making it variable as you suggested or wsa figuring out how to make it work. Will make suggested changes tho, thanks

File: projects/hslayers/src/components/styles/styler.service.ts (hunk 2 above)
Comment (FilipLeitner) [reply projects/hslayers/src/components/styles/styler.service.ts:828]: Just moved what was already used in  legend service and apparently back then when I implemented it I used date from some reason.  Will simplify this a bit.

**Comment by FilipLeitner** [comment]
> Ah, sorry, I didn't realize that this PR was meant to be merged into the hot-fix-filip branch. So I will merge into develop, but there are conflict now... 😢

Just rebased no need to fix anything

**Review by jmacura — APPROVED** [review]
Thanks!

**Pull request merged**